openmsupply/conforma-server
# #1013 Filter list improvements

Fixes #1013 

Implements both solutions as outlined in the issue.

Open to other suggestions for ways to make this operation more efficient (No, SQL "DISTINCT" won't work), although this table with over 27,000 records is about as big as it would get.

To test:

Use snapshot `/dev/snapshots/AAA_Fiji_Provisional_registration_2023_03_01_1005.zip` in templates repo, and view the "Pre-registered products" table from the "Database" menu.

Note the performance difference compared with the base branch for the following filter lists:
- "Authority" -- now uses the manually defined "New Zealand" / "Australia" list, so should be no delay
- "Route of Administration" -- the batch size difference makes the filter list load in less than a second as opposed to several seconds with the small batch size..
